tests/models/test_hooks.py

```diff
@@ -90,6 +92,58 @@ def training_epoch_end(self, outputs):
         assert metrics[f'epoch_metric_{i}'] == i
 
 
+def test_training_epoch_end_metrics_collection_on_override(tmpdir):
+    """ Test that batch end metrics are collected when training_epoch_end is overridden at the end of an epoch. """
+    num_epochs = 1
+
+    class LoggingCallback(pl.Callback):
+
+        def on_train_epoch_end(self, trainer, pl_module):
```

Comment:
@SeanNaren flake8 found this bug, but it was still merged, could pls fix it in #5666